modules/auxiliary/scanner/ntp/timeroast.rb

```diff
@@ -0,0 +1,151 @@
+##
+# This module requires Metasploit: https://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+class MetasploitModule < Msf::Auxiliary
+  include Msf::Auxiliary::Report
+  include Msf::Auxiliary::Scanner
+  include Msf::Exploit::Remote::Udp
+
+  def initialize(info = {})
+    super(
+      update_info(
+        info,
+        'Name' => 'NTP Timeroast',
+        'Description' => %q{
+          Windows authenticates NTP requests by calculating the message digest using the NT hash followed by the first
+          48 bytes of the NTP message (all fields preceding the key ID). An attacker can abuse this to recover hashes
+          that can be cracked offline for machine and trust accounts. The attacker must know the accounts RID, but
+          because RIDs are sequential, they can easily be enumerated.
+        },
+        'Author' => [
+          'Tom Tervoort',
+          'Spencer McIntyre'
+        ],
+        'License' => MSF_LICENSE,
+        'References' => [
+          ['URL', 'https://github.com/SecuraBV/Timeroast/'],
+          ['URL', 'https://www.secura.com/uploads/whitepapers/Secura-WP-Timeroasting-v3.pdf']
+        ],
+        'Notes' => {
+          'Stability' => [],
+          'Reliability' => [],
+          'SideEffects' => []
+        }
+      )
+    )
+    register_options([
+      OptIntRange.new('RIDS', [ true, 'The RIDs to enumerate (e.g. 1000-2000).' ]),
+      OptInt.new('DELAY', [ true, 'The delay in milliseconds between attempts.', 20]),
+      OptInt.new('TIMEOUT', [ true, 'The timeout in seconds to wait at the end for replies.', 5])
+    ])
+  end
+
+  def validate
+    super
+
+    errors = {}
+    errors['DELAY'] = 'DELAY can not be negative.' if datastore['DELAY'].to_i < 0
+    errors['TIMEOUT'] = 'TIMEOUT can not be negative.' if datastore['TIMEOUT'].to_i < 0
```

Comment:
It might make sense to add an `OptPositiveInt` datatore option at some point I think

Reply:
I like the idea but IMHO, it'd be better implemented with `minimum` and `maximum` arguments to `OptInt` that would be used in the `#valid?` method.

Reply:
It also works. My main concern was to avoid modules implementing the same boilerplate as you did :)